### PR TITLE
Trafodion 1822

### PR DESCRIPTION
**Pull request by gtapper**
Please publish these changes to the website. Need to build the 1.3.0 and 2.0.0 of the new Provisioning Guide.

Thanks!

### CODE REVIEW

**Comment by DaveBirdsall** [comment]
This change seems to have some conflicts?

**Comment by gtapper** [comment]
I don't see any conflicts but the message. How do I find out?

On Wed, Feb 17, 2016 at 9:33 AM, DaveBirdsall <notifications@github.com>
wrote:

> This change seems to have some conflicts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/326#issuecomment-185285861>
> .
>



-- 
Thanks,

Gunnar
*If you think you can you can, if you think you can't you're right.*

**Comment by gtapper** [comment]
Merged branches. No conflicts. Let's try again.

File: docs/provisioning_guide/src/asciidoc/_chapters/about.adoc [hunk 1]
```diff
@@ -24,19 +24,22 @@
 ////
 
 = About This Document
-This guide describes how to provision Trafodion for end-user environments. This install allows you to store and query data using Trafodion, either via Trafodion clients
+This guide describes how to provision the end-user Trafodion binaries ontop of an existing Hadoop environment. This install allows you to store and query data using Trafodion, either via Trafodion clients
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/about.adoc]: Typo, "ontop"

File: docs/provisioning_guide/src/asciidoc/_chapters/about.adoc [hunk 2]
```diff
@@ -24,19 +24,22 @@
...
 If you want to install a Trafodion developer-build environment, then please refer to the 
 http://trafodion.incubator.apache.org/contributing_redirect.html[Trafodion Contributor Guide] for instructions.
 
 == Intended Audience
-This guide is intended for:
+This guide assumes that you are well-versed in Linux and Hadoop administration. If you don't have such experience, then
+you should consider going through the steps required to install a Hadoop environment before attempting to install Trafodion.
 
-* *Single-Node Environments*: Typically used when you want to evalulate Trafodion.
+The instructions contained herein applies to the following environments.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/about.adoc]: Grammar: should be "apply"

File: docs/provisioning_guide/src/asciidoc/_chapters/activate.adoc [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+```
+
+[[activate-validate-trafodion-installation]]
+== Validate Trafodion Installation
+
+You can use sqlci (part of the base product) or trafci (requires separate install; see the
+http://trafodion.apache.org/docs/client_install/index.html[Trafodion Client Installation Guide]) to validate your installation.
+
+=== Smoke Test
+A simple smoke test to validate that Trafodion is function.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/activate.adoc]: Typo, should be "functioning".

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+The following users need be considered for Trafodion:
+
+* *Provisioning User*: A Linux-level user that performs the Trafodion provisioning tasks. This user ID
+requires `sudo` access and passwordless ssh among the nodes where Trafodion is installed. In addition,
+this user ID requires access to Hadoop distribution, HDFS, and HBase administrative users to change
+respective environment's configuration settings per Trafodion requirements. Refer to
+<<requirements-trafodion-provisioning-user,Trafodion Provisioning User>> for more information
+about the requirements and usage associated with this user ID.
+
+* *Runtime User*: A linux-level user under which the Trafodion software runs. This user ID must be registered
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Should linux be capitalized? (It is elsewhere.)

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+to Trafodion may request a user name and password, but Trafodion ignores the user name and password entered in the
+client interface, and the session runs as the database *root* user, `DB__ROOT`, without restrictions. If you want
+to restrict users, restrict access to certain users only, or restrict access to an object or operation, then you must
+enable security, which enforces authentication and authorization. Refer to
+<<enable-security,Enable Security>> for more information about this option.
+
+[[introduction-provisioning-options]]
+== Provisioning Options
+
+Trafodion ships with a set of scripts (the Trafodion Installer) that takes care of many of the installation and upgrade
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Grammar, should be "take care"

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+The Trafodion Installer supports two different modes:
+
+1. *Guided Setup*: Prompts for information as it works through the installation/upgrade process. This mode is recommended for new users.
+2. *Automated Setup*: Required information is provided in a pre-formatted bash-script configuration file, which is provided
+via a command argument when running the Trafodion Installer thereby suppressing all prompts.
++
+A template of the configuration file is available here within the installer directory: `trafodion_config_default`.
+Make a copy of the file in your directory and populate the needed information.
++
+Automated Setup is recommended since it allows you to record the required provisioning information information ahead of time.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Too bad the "Guided Setup" doesn't create a config file with the responses to all the prompts.

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc [hunk 7]
```diff
@@ -0,0 +1,466 @@
...
+```
+
+<<<
+[[introduction-trafodion-installer-install-vs-upgrade]]
+=== Install vs. Upgrade
+
+The Trafodion Installer automatically detects whether you're performing an install
+or an upgrade by looking for the Trafodion Runtime User in the `/etc/passwd` file.
+
+* If the user ID doesn't exists, then the Trafodion Installer runs in install mode.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Typo, should be "doesn't exist"

File: docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+which enables security (that is, authentication and authorization) in the Trafodion database.
+
+If you wish to manually set up the authentication configuration file and enable security, then refer to the section on
+<<enable-security,Enable Security>>.
+
+[[prepare-gather-configuration-information]]
+== Gather Configuration Information
+
+You need to gather/decide information about your environment to aid installation Trafodion, both for the Trafodion Installer
+and for recipe-based provisioning. (In alphabetical order to make it easier to find information later on in the provisioning process.)
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc]: Grammar. The parenthetical remark is not a complete sentence. Possible wordsmith: "Do so in alphabetical order..."

File: docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+
+
+<<<
+[[prepare-install-required-software-packages]]
+== Install Required Software Packages
+
+[[prepare-download-and-install-packages]]
+=== Download and Install Packages
+
+The step is required if you're:
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc]: Do you mean, "This step..."?

File: docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+* Installing Trafodion on SUSE.
+* Using Recipe-Based Provisioning.
+* Can't download the required software packages using the Internet.
+
+If none of these situations exist, then we highly recommend that you use the Trafodion Installer.
+
+You perform this step as a user with `root` or `sudo` access.
+
+Install the packages listed in <<requirements-software-packages,Software Packages>> above on all nodes in the cluster. Note the special
+handling for `log4c&#43;&#43;`. See <<prepare-build-and-install-log4cplusplus, Build and Install log4c++>> below for more information.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc]: Three different ways to render "++"! :-)

File: docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc [hunk 11]
```diff
@@ -345,29 +386,28 @@ The following changes are required in `hdfs-site.xml`:
 | dfs.namenode.acls.enabled | true | Enable HDFS  POSIX Access Control Lists (ACLs).
 |===
 
+[[requirements-hbase-changes]]
 === HBase Changes
 
-NOTE: These changes require a restart of ZooKeeper on all nodes in the cluster.
+NOTE: These changes require a restart of ZooKeeper HBase on all nodes in the cluster.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc]: Do you mean "ZooKeeper and HBase"?

File: docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc [hunk 12]
```diff
@@ -415,11 +456,34 @@ java.heapsize^1^ | 1073741824 (1GB) | Java Heap Size for the HDFS Secondary Name
...
 == Recommended Configuration Changes
 The following configuration changes are recommended but not required.
 
 NOTE: The Trafodion Installer does *not* make these changes.
 
+[[requirements-recommended-security-changes]]
+=== Recommended Security Changes
+
+The `trafodion` user ID should not be given other `sudo` privileges than what's specified in this manual. Also, we
+that this user ID is locked (`sudo passwd -l trafodion`) once the installation/upgrade activity has been completed.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc]: Missing verb? Did you mean, "Also, we checked that this user ID is locked..." Or perhaps you meant, "The Trafodion Installer checks that this user ID is locked..."

File: docs/provisioning_guide/src/asciidoc/_chapters/script_install.adoc [hunk 13]
```diff
@@ -0,0 +1,511 @@
...
+```
+
+[[install-automated-install]]
+== Automated Install
+
+The `--config_file` option runs the Trafodion in Automated Setup mode. Refer to <<introduction-trafodion-installer,Trafodion Installer>>
+in the <<introduction,Introduction>> chapter for instructions of how you edit your configuration file.
+
+Edit your config file using the information you collected in the <<prepare-gather-configuration-information,Gather Configuration Information>>
+in the <<prepare,Prepare>> chapter.
```
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/script_install.adoc]: Do you want to say, "step" before "in the <<prepare,Prepare>> chapter"?

**Comment by DaveBirdsall** [comment]
Looks good! This was quite a lot of work. Thanks too for supplying the missing egg logo.

File: docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc (hunk 10 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc]: Yep. Markup-language limitations.

On Wed, Feb 17, 2016 at 1:17 PM, DaveBirdsall <notifications@github.com>
wrote:

> In docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc
> <https://github.com/apache/incubator-trafodion/pull/326#discussion_r53224885>
> :
>
> > +
> > +[[prepare-download-and-install-packages]]
> > +=== Download and Install Packages
> > +
> > +The step is required if you're:
> > +
> > +* Installing Trafodion on SUSE.
> > +* Using Recipe-Based Provisioning.
> > +* Can't download the required software packages using the Internet.
> > +
> > +If none of these situations exist, then we highly recommend that you use the Trafodion Installer.
> > +
> > +You perform this step as a user with `root` or `sudo` access.
> > +
> > +Install the packages listed in <<requirements-software-packages,Software Packages>> above on all nodes in the cluster. Note the special
> > +handling for `log4c&#43;&#43;`. See <<prepare-build-and-install-log4cplusplus, Build and Install log4c++>> below for more information.
>
> Three different ways to render "++"! :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/326/files#r53224885>.
>



-- 
Thanks,

Gunnar
*If you think you can you can, if you think you can't you're right.*

File: docs/provisioning_guide/src/asciidoc/_chapters/about.adoc (hunk 1 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/about.adoc]: Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/about.adoc (hunk 2 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/about.adoc]: Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/activate.adoc (hunk 3 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/activate.adoc]: Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc (hunk 5 above)
Comment (DaveBirdsall) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: My comment is incorrect; "takes care" is correct. ("Takes care" refers to "set")

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc (hunk 4 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Good catch. Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc (hunk 5 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc (hunk 6 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: Yep. We have a plan to add more discovery to reduce the number of items in the config file. 2.0 maybe.

File: docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc (hunk 7 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/introduction.adoc]: It should. Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc (hunk 8 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc]: Listed in... fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc (hunk 9 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/prepare.adoc]: Absolutely. Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc (hunk 11 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc]: I did. Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc (hunk 12 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/requirements.adoc]: We recommend. :)

Fixed.

File: docs/provisioning_guide/src/asciidoc/_chapters/script_install.adoc (hunk 13 above)
Comment (gtapper) [inline docs/provisioning_guide/src/asciidoc/_chapters/script_install.adoc]: I do! Fixed.

**Comment by gtapper** [comment]
Thanks for all the comments. It's easy to get blind in text files. I'll check in a new version and then move onto simplification. The theory is that a lot of the requirements are fulfilled when installing Hadoop. So, I'll test that and then we'll know what can be simplified.

**Comment by DaveBirdsall** [comment]
I'll merge this tomorrow morning if there are no further comments or changes.

**Pull request merged**